### PR TITLE
API Refactor CKAN API client and resource population to support metadata retrieval

### PR DESCRIPTION
**Pull request by robbieaverill**
* Allows fetching of the dataset metadata as well as the resource data (fields)
* Displays the dataset and resource name as the GridField heading for columns (when available)

Resolves https://github.com/silverstripe/silverstripe-ckan-registry/issues/81

### CODE REVIEW

File: src/Service/Client.php [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+namespace SilverStripe\CKANRegistry\Service;
+
+use GuzzleHttp\Psr7\Request;
+use RuntimeException;
+use SilverStripe\CKANRegistry\Model\Resource;
+use SilverStripe\Core\Extensible;
+use SilverStripe\Core\Injector\Injectable;
+
+class Client implements ClientInterface
```
Comment (ScopeyNZ) [inline src/Service/Client.php]: Is there a better name than `SilverStripe\CKANRegistry\Service\Client`? It feels a little too generic.

Comment (ScopeyNZ) [reply src/Service/Client.php]: I get that `CKANRegistry` is in the namespace but it still doesn't seem very clear that this is a base API client for CKAN specifically.

Comment (robbieaverill) [reply src/Service/Client.php]: How about APIClient? I originally had CKAN in there but it's also in the namespace so I removed it

Comment (ScopeyNZ) [reply src/Service/Client.php]: I'd be happy with `APIClient`

**Pull request merged**